service/controller/cluster.go

```diff
@@ -130,18 +147,250 @@ func newClusterResourceSets(config ClusterConfig) ([]*controller.ResourceSet, er
+		}
+
+		ops, err := configmap.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		configMapResource, err = toCRUDResource(config.Logger, ops)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var deploymentResource resource.Interface
+	{
+		c := deployment.Config{
+			DNSServers:    config.DNSServers,
+			G8sClient:     config.K8sClient.G8sClient(),
+			K8sClient:     config.K8sClient.K8sClient(),
+			Logger:        config.Logger,
+			NTPServers:    config.NTPServers,
+			TenantCluster: config.TenantCluster,
+		}
+
+		ops, err := deployment.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		deploymentResource, err = toCRUDResource(config.Logger, ops)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var ingressResource resource.Interface
+	{
+		c := ingress.Config{
+			K8sClient: config.K8sClient.K8sClient(),
+			Logger:    config.Logger,
+		}
+
+		ops, err := ingress.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		ingressResource, err = toCRUDResource(config.Logger, ops)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var nodeIndexStatusResource resource.Interface
+	{
+		c := nodeindexstatus.Config{
+			G8sClient: config.K8sClient.G8sClient(),
+			Logger:    config.Logger,
+		}
+
+		nodeIndexStatusResource, err = nodeindexstatus.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var pvcResource resource.Interface
+	{
+		c := pvc.Config{
+			K8sClient: config.K8sClient.K8sClient(),
+			Logger:    config.Logger,
+		}
+
+		ops, err := pvc.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		pvcResource, err = toCRUDResource(config.Logger, ops)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var serviceResource resource.Interface
+	{
+		c := service.Config{
+			K8sClient: config.K8sClient.K8sClient(),
+			Logger:    config.Logger,
+		}
+
+		ops, err := service.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		serviceResource, err = toCRUDResource(config.Logger, ops)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var statusResource resource.Interface
+	{
+		c := statusresource.ResourceConfig{
+			ClusterEndpointFunc:      key.ToClusterEndpoint,
+			ClusterIDFunc:            key.ToClusterID,
+			ClusterStatusFunc:        key.ToClusterStatus,
+			NodeCountFunc:            key.ToNodeCount,
+			Logger:                   config.Logger,
+			RESTClient:               config.K8sClient.G8sClient().ProviderV1alpha1().RESTClient(),
+			TenantCluster:            config.TenantCluster,
+			VersionBundleVersionFunc: key.ToOperatorVersion,
 		}
 
-		resourceSet, err = NewClusterResourceSet(c)
+		statusResource, err = statusresource.NewResource(c)
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
 	}
 
-	resourceSets := []*controller.ResourceSet{
-		resourceSet,
+	resources := []resource.Interface{
```

Comment:
I assume you checked that this order is unchanged - it's a bit hard to crosscheck in the diff.

Reply:
It's unchanged. I am feeling less happy with this now because it bloats the PR and isn't entirely necessary. I will revert this change and move it into a separate refactoring PR.

Reply:
This is reverted. Should be easier to see changes.